lib/utils.py

```diff
@@ -1,16 +1,30 @@
 # Functionality that will be shared by more than one library
 
-import os
 import math
+import os
+import re
 
 
 def copy_to_clipboard(string_to_copy):
     try:
         import pyperclip
+
         pyperclip.copy(string_to_copy)
     except ImportError:
         # Don't crash if pyperclip not installed
-        log("Pyperclip not installed. Clipboard related functionality won't work", LogUrgency.WARNING)
+        log(
+            "Pyperclip not installed. Clipboard related functionality won't work",
+            LogUrgency.WARNING,
+        )
+
+
+def validate_url(url: str) -> bool:
```

Comment:
I prefer that functions that return `bool` have names starting with `is_`. This produces better and more understandable code in a bigger codebase. so `validate_url` -> `is_url_valid`